recovery/src/main/java/com/kin/ecosystem/recovery/events/BroadcastManagerImpl.java

```diff
@@ -51,6 +51,7 @@ public void unregister() {
 		if (receiver != null) {
 			LocalBroadcastManager.getInstance(activity).unregisterReceiver(receiver);
 		}
+		activity = null;
```

Comment:
how can you be sure that BroadcastManagerImpl is not reused - meaning same BroadcastManagerImpl is registered to,  unregister and register again

Reply:
activity is also used in more places and not protected against NPE

Reply:
1. The unregister is being call from `BackupManager.release()` meaning the you are not going to use the same `BackupManager` again, so it should be initialized again for use.
2. where ?

Reply:
Could happen I how could you be sure it will not happen in the future